localstack/utils/auth.py

```diff
@@ -0,0 +1,39 @@
+import logging
+
+from botocore.auth import HmacV1Auth
+
+logger = logging.getLogger(__name__)
+
+
+class HmacV1QueryAuth(HmacV1Auth):
+    """
+    Generates a presigned request for s3.
+
+    Spec from this document:
+
+    http://docs.aws.amazon.com/AmazonS3/latest/dev/RESTAuthentication.html
+    #RESTAuthenticationQueryStringAuth
+
+    """
+    DEFAULT_EXPIRES = 3600
+
+    def __init__(self, credentials, expires=DEFAULT_EXPIRES):
+        self.credentials = credentials
+        self._expires = expires
+
+    def _get_date(self):
+        return str(int(int(self._expires)))
+
+    def get_signature(self, string_to_sign):
+        return self.sign_string(string_to_sign)
+
+    def get_string_to_sign(self, method, split, headers, expires=None,
+                      auth_path=None):
+        if self.credentials.token:
+            del headers['x-amz-security-token']
```

Comment:
nit: `del` should not be required, as we're overwriting the value in the next line.

Reply:
You are correct